tests/tests_quadlet_demo.yml

```diff
@@ -0,0 +1,171 @@
+    # healthcheck, Secret= directive.
+    # NOTE: EL8 copr currently not working
+    podman_use_copr: false  # disable copr for CI testing
+    podman_fail_if_too_old: false
+    podman_create_host_directories: true
+    podman_activate_systemd_unit: false
+    podman_quadlet_specs:
+      - file_src: quadlet-demo.network
+      - file_src: quadlet-demo-mysql.volume
+      - template_src: quadlet-demo-mysql.container.j2
+      - file_src: envoy-proxy-configmap.yml
+      - file_src: quadlet-demo.yml
+      - file_src: quadlet-demo.kube
+        activate_systemd_unit: true
+    podman_firewall:
+      - port: 8000/tcp
+        state: enabled
+      - port: 9000/tcp
+        state: enabled
+    __test_cert_name: quadlet_demo
+    podman_host_directories:
+      "/tmp/quadlet_demo":
+        mode: "0777"
+  tasks:
+    - name: Generate certificates
+      include_role:
+        name: fedora.linux_system_roles.certificate
+      vars:
+        certificate_requests:
+          - name: "{{ __test_cert_name }}"
+            dns: ["localhost"]
+            ca: self-sign
+        certificate_test_mode: true
+        certificate_test_remove_files: true
+
+    - name: Create random string and save it in root_password
+      set_fact:
+        root_password: "{{ lookup('lines', 'openssl rand -base64 48') }}"
+
+    - name: Run the role
+      include_role:
+        name: linux-system-roles.podman
+      vars:
+        podman_secrets:
+          - name: mysql-root-password-container
+            state: present
+            skip_existing: true
+            data: "{{ root_password }}"
+          - name: mysql-root-password-kube
+            state: present
+            skip_existing: true
+            data: |
+              apiVersion: v1
+              data:
+                password: "{{ root_password | b64encode }}"
+              kind: Secret
+              metadata:
+                name: mysql-root-password-kube
+          - name: envoy-certificates
+            state: present
+            skip_existing: true
+            data: |
+              apiVersion: v1
+              data:
+                certificate.key: {{
+                  certificate_test_certs[__test_cert_name]['key_content'] |
+                  b64encode }}
+                certificate.pem: {{
+                  certificate_test_certs[__test_cert_name]['cert_content'] |
+                  b64encode }}
+              kind: Secret
+              metadata:
+                name: envoy-certificates
+
+    - name: Check
+      command: ls -alrtF /etc/containers/systemd
+      changed_when: false
+
+    - name: Check containers
+      command: podman ps -a
+      changed_when: false
+      failed_when: false
+
+    - name: Check pods
+      command: podman pod ps --ctr-ids --ctr-names --ctr-status
+      changed_when: false
+      failed_when: false
+
+    - name: Check systemd
+      # noqa command-instead-of-module
+      shell: set -euo pipefail; systemctl list-units | grep quadlet
```

Comment:
Quadlet is a generator executed by systemd, it is not a unit and therefore this check actually fails.
This also raises the question of what it the purpose of of these steps if `failed_when` is set to `false`

Reply:
> Quadlet is a generator executed by systemd, it is not a unit and therefore this check actually fails.

Hmm - well it does print something.

> This also raises the question of what it the purpose of of these steps if `failed_when` is set to `false`

This isn't really an assertion test - it's really so I can see what's going on - I guess I should convert this into a "real" test, or just get rid of it, once the quadlet stuff is working and reviewed.